safehttp/plugins/xsrf/xsrf_test.go

```diff
@@ -87,164 +68,184 @@ func TestXSRFTokenPost(t *testing.T) {
 			wantBody: "Forbidden\n",
 		},
 	}
-	for _, test := range tests {
-		i := xsrf.Interceptor{AppKey: "xsrf", Identifier: userIdentifier{}}
-		tok := xsrftoken.Generate("xsrf", test.userID, test.host+test.path)
-		req := safehttptest.NewRequest("POST", "http://foo.com/pizza", strings.NewReader(xsrf.TokenKey+"="+tok))
-		req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+)
 
-		rec := safehttptest.NewResponseRecorder()
-		i.Before(rec.ResponseWriter, req)
+func TestTokenPost(t *testing.T) {
+	for _, method := range stateChangingMet {
+		for _, test := range formTokenTests {
+			t.Run(method+" "+test.name, func(t *testing.T) {
+				i := Interceptor{AppKey: "xsrf", Identifier: userIdentifier{}}
+				tok := xsrftoken.Generate("xsrf", test.userID, test.actionID)
+				req := safehttptest.NewRequest(method, "https://foo.com/pizza", strings.NewReader(TokenKey+"="+tok))
+				req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+				rec := safehttptest.NewResponseRecorder()
 
-		if rec.Status() != test.wantStatus {
-			t.Errorf("response status: got %v, want %v", rec.Status(), test.wantStatus)
-		}
-		if diff := cmp.Diff(test.wantHeader, map[string][]string(rec.Header())); diff != "" {
-			t.Errorf("rec.Header() mismatch (-want +got):\n%s", diff)
-		}
-		if got := rec.Body(); got != test.wantBody {
-			t.Errorf("response body: got %q want %q", got, test.wantBody)
+				i.Before(rec.ResponseWriter, req)
```

Comment:
This is a nit, but try to somehow sort the unrelated lines. What would probably look better is:

1. Creating the Response Recorder.
2. Creating the token, creating the request.
3. Creating the interceptor and running it on the request.

This way you minimize the context needed for each line of code to be able to understand it.